proposals/devcontainer-features.md

```diff
@@ -0,0 +1,143 @@
+# Dev container features reference
+
+Dev container features provide a smooth path for customizing your container definitions. Features are self contained units of code meant to facilitate creation of custom images for application or development containers.
+
+From a practical point of view, features are folders that contain units of code with different entrypoints for different lifecycle events.
+
+Features can be defined by a `feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.
+
+Features are to be executed in sequence as defined in `devcontainer.json`.
+
+## Folder Structure 
+
+A feature is a self contained entity in a folder. A feature release would be a tar file that contains all files part of the feature.
+
++-- feature
+|    +-- feature.json
+|    +-- validate.sh (default)
+|    +-- aquire.sh (default)
+|    +-- install.sh (default)
+|    +-- (other files)
+
+In case `feature.json`  does not include a reference for the lifecycle scripts the application will look for the default script names and will execute them if available.
+
+In case there is intent to create a set of features that share code, it is possible to create a feature collection in the following way:
+
+collectionFolder
++-- feature-collection.json
++-- common (or similar)
+|    +-- (library files)
++-- feature1
+|    +-- feature.json
+|    +-- install.sh (or others)
+|    +-- (other files)
++-- feature2
+(etc)
+
+## feature.json properties
+
+the `feature.json` file defines information about the feature to be used by any tool that helps a user utilize them, and the way the feature will be executed.
+
+The properties of the file are as follows:
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The id should be unique in the context of the repository/published package where they exist. |
+| name | string | Name of the feature/definition |
+| description | string | Description of the feature/definition |
+| documentationURL | string | Url that targets the documentation of the feature. |
+| licenseURL | string | Url that points to the license of the feature. |
```

Comment:
Can we add an [OSI boolean](https://opensource.org/licenses)?
I will be faster to filter then regex on license remote files.

Reply:
The scope was related to having something machine readable so that we can quckly filter also in the UI.

How can I quickly filter OSS only elements?

Reply:
@edgonmsft Do I need to open a new ticket to track this?

Reply:
Yep, lets create a new one. I think its a good idea the we should flesh out a little bit more. Thanks!